src/settings.rs

```diff
@@ -275,4 +285,27 @@ where
     fn interface_mut(&mut self) -> &mut Self::Interface {
         &mut self.interface
     }
+
+    fn clear(&mut self, buf: &mut [u8], key: Option<&str>) {
+        let range = self.storage.range();
+        if let Some(key) = key {
+            embassy_futures::block_on(sequential_storage::map::remove_item::<
+                SettingsItem,
+                _,
+            >(
+                &mut self.storage,
+                range,
+                sequential_storage::cache::NoCache::new(),
+                buf,
+                heapless::String::from(key),
+            ))
+            .unwrap()
+        } else {
+            // Erase the whole flash range.
+            embassy_futures::block_on(
+                self.storage.erase(range.start, range.end),
```

Comment:
Yes, this erases 1MB of flash and takes quite a while to do (several seconds).

I definitely think this could be done more efficiently on the sequential-storage side of things (i.e. scan the device, look for used pages, erase those pages only). I'll take a look.

Reply:
We should also reduce the flash range for sequential storage. We don't need to allocate 1 MB.

Reply:
It can do with 2 right now. But your effective storage size `page_size * (num_pages - 1)` for map, because it needs to keep one page as a buffer page.

Reply:
Ack. Nice work on sequential-storage!
Let's just take current sequential storage master, wait for release, and 2 pages (three breaking changes in one ;). It'll be a while in practice until it needs to erase pages.

Reply:
Yeah I'm currently testing, but hitting HardFault exceptions after erasing and then re-reading flash sections. I believe it's due to flash ECC code failures, but what we're doing _should_ be supported (since we're only clearing bits). I'll dig into what's going wrong here over the next day or two

Reply:
The hard faults are indeed being caused by flash ECC failures. Refer to the analysis in https://github.com/quartiq/stabilizer/pull/884#discussion_r1644776679